nextcord/channel.py

```diff
@@ -753,6 +769,428 @@ async def create_thread(
+            If ``None``, retrieves every archived thread in the channel. Note, however,
+            that this would make it a slow operation.
+        before: Optional[Union[:class:`abc.Snowflake`, :class:`datetime.datetime`]]
+            Retrieve archived channels before the given date or ID.
+        private: :class:`bool`
+            Whether to retrieve private archived threads.
+        joined: :class:`bool`
+            Whether to retrieve private archived threads that you've joined.
+            You cannot set ``joined`` to ``True`` and ``private`` to ``False``.
+
+        Raises
+        ------
+        Forbidden
+            You do not have permissions to get archived threads.
+        HTTPException
+            The request to get the archived threads failed.
+
+        Yields
+        -------
+        :class:`Thread`
+            The archived threads.
+        """
+
+        return ArchivedThreadIterator(
+            self.id, self.guild, limit=limit, joined=joined, private=private, before=before
+        )
+
+
+class ForumChannel(abc.GuildChannel, Hashable):
+    # TODO: docstring
+    """Represents a Discord guild forum channel.ster
+
+    .. container:: operations
+
+        .. describe:: x == y
+
+            Checks if two channels are equal.
+
+        .. describe:: x != y
+
+            Checks if two channels are not equal.
+
+        .. describe:: hash(x)
+
+            Returns the channel's hash.
+
+        .. describe:: str(x)
+
+            Returns the channel's name.
+    """
+
+    __slots__ = (
+        "id",
+        "guild",
+        "name",
+        "category_id",
+        "position",
+        "topic",
+        "nsfw",
+        "flags",
+        "slowmode_delay",
+        "default_auto_archive_duration",
+        "last_message_id",
+        "_state",
+        "_type",
+        "_overwrites",
+    )
+
+    def __init__(self, *, state: ConnectionState, guild: Guild, data: ForumChannelPayload):
+        self._state: ConnectionState = state
+        self.id: int = int(data["id"])
+        self._type: int = data["type"]
+        self._update(guild, data)
+
+    def _update(self, guild: Guild, data: ForumChannelPayload) -> None:
+        self.guild = guild
+        self.name = data["name"]
+        self.category_id: Optional[int] = utils._get_as_snowflake(data, "parent_id")
+        self.topic: Optional[str] = data.get("topic")
+        self.position: int = data["position"]
+        self.nsfw: bool = data.get("nsfw", False)
+        # Does this need coercion into `int`? No idea yet.
+        self.slowmode_delay: int = data.get("rate_limit_per_user", 0)
+        self.flags: ChannelFlags = ChannelFlags._from_value(data.get("flags", 0))
+        self.default_auto_archive_duration: ThreadArchiveDuration = data.get(
+            "default_auto_archive_duration", 1440
+        )
+        self.last_message_id: Optional[int] = utils._get_as_snowflake(data, "last_message_id")
+        self._fill_overwrites(data)
```

Comment:
unfortunately `_update` is used by existing events to update classes, this would need a refactor of those which is outside of this PR

https://canary.discord.com/channels/881118111967883295/881124672056467456/990736744758509578